includes/Modules/Sign_In_With_Google/Web_Tag.php

```diff
@@ -0,0 +1,215 @@
+			const res = await fetch( '<?php echo esc_js( $login_uri ); ?>', {
+				method: 'POST',
+				headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+				body: new URLSearchParams( response )
+			} );
+
+			<?php if ( empty( $redirect_to ) && ! $is_login_page && $should_show_one_tap_prompt ) : // phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect ?>
+				location.reload();
+			<?php else : // phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect ?>
+				if ( res.ok && res.redirected ) {
+					location.assign( res.url );
+				}
+			<?php endif; // phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect ?>
+		} catch( error ) {
+			console.error( error );
+		}
+	}
+
+	google.accounts.id.initialize( {
+		client_id: '<?php echo esc_js( $settings['clientID'] ); ?>',
+		callback: handleCredentialResponse,
+		library_name: 'Site-Kit'
+	} );
+
+	<?php if ( $is_wp_login ) : // phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect ?>
+		const buttonDivToAddToLoginForm = document.createElement( 'div' );
+		buttonDivToAddToLoginForm.classList.add( 'googlesitekit-sign-in-with-google__frontend-output-button' );
+
+		document.getElementById( 'login' ).insertBefore( buttonDivToAddToLoginForm, document.getElementById( 'loginform' ) );
+	<?php endif; // phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect ?>
+
+	<?php if ( ! is_user_logged_in() || $is_wp_login ) : // phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect ?>
+			<?php
+			/**
+			 * Render SiwG buttons for all `<div>` elements with the "magic
+			 * class" on the page.
+			 *
+			 * Mainly used by Gutenberg blocks.
+			 */
+			?>
+		document.querySelectorAll( '.googlesitekit-sign-in-with-google__frontend-output-button' ).forEach( ( siwgButtonDiv ) => {
+			google.accounts.id.renderButton( siwgButtonDiv, <?php echo wp_json_encode( $btn_args ); ?> );
+		});
+	<?php endif; // phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect ?>
+
+	<?php if ( $should_show_one_tap_prompt ) : // phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect ?>
+		google.accounts.id.prompt();
+	<?php endif; // phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect ?>
+
+	<?php if ( ! empty( $redirect_to ) ) : // phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect ?>
+		const expires = new Date();
+		expires.setTime( expires.getTime() + <?php echo esc_js( $cookie_expire_time ); ?> );
+		document.cookie = "<?php echo esc_js( Authenticator::COOKIE_REDIRECT_TO ); ?>=<?php echo esc_js( $redirect_to ); ?>;expires=" + expires.toUTCString() + ";path=<?php echo esc_js( Authenticator::get_cookie_path() ); ?>";
+	<?php endif; // phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect ?>
+} )();
+		<?php
+
+		// Strip all whitespace and unnecessary spaces.
+		$inline_script = preg_replace( '/\s+/', ' ', ob_get_clean() );
+		$inline_script = preg_replace( '/\s*([{};\(\)\+:,=])\s*/', '$1', $inline_script );
+
+		// Output the Sign in with Google script.
+		print( "\n<!-- Sign in with Google button added by Site Kit -->\n" );
```

Comment:
Please output comments as we do in other places for consistency. The text should also be translated.

Reply:
Updated following the AdSense tag pattern:

https://github.com/google/site-kit-wp/blob/9e2ae25506a5c13d100af03c6b1919639a2a251e/includes/Modules/AdSense/Web_Tag.php#L95-L97